### DISCUSSION TITLE
[Suggestion / Request] Item - Hasitem - ItemRequirement System upgrade

### DISCUSSION THREAD

**Discussion by Angelfox28** [Zrips/CMI#1344] [Enhancement]
At the moment, only in the kits system you can insert objects with custom name, descriptions and enchantments.
I quote an example

> ```
> Items: 
>     '1': 
>       v: 1631
>       type: STONE
>       amount: 64
>       meta:
>         ==: ItemMeta
>         meta-type: UNSPECIFIC
>         display-name: &3Stone
>         lore:
>         - &6Line 1 
>         - &7Test
>         - &4Fire
>         - &9Ice
>         enchants:
>           THORNS: 11
>           ARROW_DAMAGE: 3
>         ItemFlags:
>         - HIDE_ENCHANTS
> ```

It would be nice to be able to do it also in the following functionalities: Interactive command, ranks, portals and schedulers,to improve the following Specialized Command
**item: [item: (date) (- amount)]!
hasitem: [item: (date) (- amount)]!**
in order to check if players have such particular items with descriptions or enchantments

Example for create an IC with item request:
When creating an IC, you must have the object in your hand, so in the interactiveCommand.yml the information of the object is directly written.
Possible command:
**Hasitem:[amount]!** cmi [playerName] fly

**Comment by Zrips** [comment]
Kits has ingame editor which converts items into text format autocratically and that could be quite long entry. But that aside, currently you can only check by item material type and amount, which is in most cases more then enough. We could expand on this at some point to check for items by some sub data, like name, lore, enchants and NBT data values.

**Comment by Angelfox28** [comment]
It would be great because in the server there are many items with names, descriptions custom and enchantments.
Thanks
